src/MutableArithmetics.jl

```diff
@@ -1,5 +1,133 @@
+#  Copyright 2019, Gilles Peiffer, Benoît Legat, Sascha Timme, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#############################################################################
+
 module MutableArithmetics
 
-greet() = print("Hello World!")
+using LinearAlgebra, Base.GMP.MPZ
+
+import Base.+, Base.*, Base.==
+import LinearAlgebra.mul!
+
+export DummyBigInt
+
+
+"""
+    AbstractMutable
+
+Generic supertype for types
+implementing the MutableArithmetics interface.
+"""
+abstract type AbstractMutable end
+
+"""
+    DummyBigInt
+
+Mutable wrapper type around `BigInt`.
+The goal of this type is to allow the package to test itself;
+hence its name.
+"""
+struct DummyBigInt <: AbstractMutable
+    data::BigInt
+end
+
+# For convenience.
+Base.zero(::Type{DummyBigInt}) = DummyBigInt(0)
+Base.convert(::Type{DummyBigInt}, x::Int64) = DummyBigInt(x)
+
+# Arithmetics
+*(a::DummyBigInt, b::DummyBigInt) = DummyBigInt(a.data * b.data)
```

Comment:
I tend to prefer not importing `*` and doing using `Base.:*` here so that everything is in one place but that's a matter of taste :)

Reply:
I prefer that too, but didn't know about that syntactic sugar. I'll change it, it'll also give me the chance to check the CI/coverage plugins are all running now.